sdk/core/core-http/test/proxyAgent.node.ts

```diff
@@ -63,6 +63,47 @@ describe("proxyAgent", () => {
       agent.proxyOptions.headers!.should.contain({ "user-agent": "Node.js" });
       done();
     });
+
+    [
+      { host: "host", port: 0 },
+      { host: "host", port: 65535 }
+    ].forEach((testCase) => {
+      it(`should not throw error when being given a valid proxy settings of { host: '${testCase.host}', port: ${testCase.port} }.`, function(done) {
+        const proxySettings = {
+          host: testCase.host,
+          port: testCase.port
+        };
+
+        const fn = function() {
+          createProxyAgent("http://example.com", proxySettings);
+        };
+        fn.should.not.throw();
+        done();
+      });
+    });
+
+    [
+      { host: "", port: 8080, expectInvalidHostError: true },
```

Comment:
@jeremymeng Can you add the test case of `host` being `null` or `undefined` as well?

Reply:
`host` is defined as of type `string` so we should be good

```typescript
/**
 * Options to configure a proxy for outgoing requests (Node.js only).
 */
export interface ProxySettings {
  /*
   * The proxy's host address.
   */
  host: string;

  /*
   * The proxy host's port.
   */
  port: number;

  /**
   * The user name to authenticate with the proxy, if required.
   */
  username?: string;

  /**
   * The password to authenticate with the proxy, if required.
   */
  password?: string;
}

```